newrelic-agent/src/main/java/com/newrelic/agent/config/AgentConfig.java

```diff
@@ -179,6 +179,11 @@ public interface AgentConfig extends com.newrelic.api.agent.Config, DataSenderCo
      */
     ApplicationLoggingConfig getApplicationLoggingConfig();
 
+    /**
+     * Get the Code Level Metrics config.
+     */
+    CodeLevelMetricsConfig getCodeLevelMetricsConfig();
```

Comment:
I noticed sometimes we have added new config to the interface and some have only been added to the AgentConfigImpl.  I'm just curious for my own knowledge what is the determining factor on whether to include it here.  Also there is a confusing comment on the class which I've never fully understood why it was there:
```
* Adding new methods to this is no longer preferred. Please access new config using the {@link com.newrelic.api.agent.Config#getValue(String)}
 * method with the keyPath being '.' delimited hierarchy of keys, similar to the command line setting override.<br/>. 
```
Does it apply here or is this comment itself outdated?

Reply:
It should be added to the interface so we don't rely on the implementation details. This is specifically useful for testing, where we could have a AgentConfigStub that implements AgentConfig. Then if all the code only uses AgentConfig, then we can easily switch to the stub for testing.

I keep forgetting about that comment.
I followed the Logging configuration, which was the last big piece of configuration we added.
I don't really like that comment, to tell you the truth. I like the fact that you call something like `agentConfig.getLoggingConfig().isEnabled`. Or maybe it is just that I really don't like passing Strings to methods (when enums would be a better choice).